[program.cpp:330] | (T3K) T3000 unit tests | t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] T3K ttnn tests fail with kernel compile-time static_assert 'Index out of range' (41 < 41) in minimal_default_writer / all_gather_async
<!-- AUTO-TRIAGE-SUMMARY-END -->
**Workflow:** (T3K) T3000 unit tests
**Job:** t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]

### Regression timeline
- Last passing commit: N/A
- First failing commit: `62a086c55fa7` — 2026-08-05T18:30:22Z — https://github.com/tenstorrent/tt-metal/actions/runs/31035040690/job/92407653751
- Most recent failing commit: `254df50022a9` — 2026-08-06T00:22:02Z — https://github.com/tenstorrent/tt-metal/actions/runs/31059470614/job/92486212794

### Failing job URLs (last 2 runs)
- https://github.com/tenstorrent/tt-metal/actions/runs/31059470614/job/92486212794
- https://github.com/tenstorrent/tt-metal/actions/runs/31035040690/job/92407653751

### Failing test path(s)
- t3000-unit-tests / t3k_ttnn_tests [wh_llmbox] — kernel build for `ttnn/cpp/ttnn/operations/experimental/ccl/all_gather_async/device/kernels/minimal_default_writer.cpp` (brisc firmware compile step, via `tt::tt_metal::JitBuildState::compile_one` / program.cpp:330)

### Error excerpt (terminal failure from logs)
```
C++ exception with description "TT_THROW @ /work/tt_metal/impl/program/program.cpp:330: tt::exception
info:
Failed to generate binaries for minimal_default_writer TT_THROW @ /work/tt_metal/jit_build/build.cpp:70: tt::exception
info:
brisc build failed. Log: In file included from /work/tt_metal/hw/inc/internal/firmware_common.h:14,
                 from /work/tt_metal/hw/firmware/src/tt-1xx/brisck.cc:17:
/work/tt_metal/hw/inc/api/compile_time_args.h: In instantiation of 'constexpr uint32_t get_ct_arg() [with long unsigned int Idx = 41; uint32_t = long unsigned int]':
/work/tt_metal/hw/inc/api/tensor/tensor_accessor_args.h:58:20:   required from 'constexpr const uint32_t TensorAccessorArgs<40>::AlignedPageSize'
/work/ttnn/cpp/ttnn/operations/experimental/ccl/all_gather_async/device/kernels/minimal_default_writer.cpp:170:82:   required from here
  170 |     const auto output_addrgen = TensorAccessor(output_tensor_args, output_address);
/work/tt_metal/hw/inc/api/compile_time_args.h:27:23: error: static assertion failed: Index out of range
   27 |     static_assert(Idx < kernel_compile_time_args.size(), "Index out of range");
/work/tt_metal/hw/inc/api/compile_time_args.h:27:23: note: the comparison reduces to '(41 < 41)'
```

### Reproduction steps
1. Run the T3000 unit test job `t3k_ttnn_tests` on a wh_llmbox machine (per `.github/workflows` T3K T3000 unit tests config), which builds and runs ttnn CCL tests exercising `all_gather_async`.
2. Alternatively, build the `minimal_default_writer` kernel directly by running any ttnn test that instantiates `all_gather_async` (device/kernels/minimal_default_writer.cpp), which triggers JIT compilation of the brisc firmware with the current compile-time-args count for `TensorAccessor`/`TensorAccessorArgs`.
3. Observe the brisc compile fails with `static_assert(Idx < kernel_compile_time_args.size())` reducing to `(41 < 41)`, i.e., the code requests compile-time-arg index 41 while only 41 (indices 0-40) are available — an off-by-one/out-of-range access when building `TensorAccessorArgs<40>::AlignedPageSize`.

### Notes
- This is a compile-time (build) failure, not a runtime assertion or timeout — the kernel source references a compile-time-arg index that exceeds the number of args provided.
- Determinism check:
  - Run 1 (job 92486212794): terminal error — `/work/tt_metal/hw/inc/api/compile_time_args.h:27:23: error: static assertion failed: Index out of range` / `the comparison reduces to '(41 < 41)'`, from `minimal_default_writer.cpp:170:82`.
  - Run 2 (job 92407653751): terminal error — identical: `/work/tt_metal/hw/inc/api/compile_time_args.h:27:23: error: static assertion failed: Index out of range` / `the comparison reduces to '(41 < 41)'`, from `minimal_default_writer.cpp:170:82`, wrapped in `TT_THROW @ /work/tt_metal/impl/program/program.cpp:330` and `Failed to generate binaries for minimal_default_writer`.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (T3K) T3000 unit tests`
`Auto-triage-job-name: t3000-unit-tests / t3k_ttnn_tests [wh_llmbox]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/tests/ttnn/unit_tests/gtests/test_generic_op.cpp
+++ b/tests/ttnn/unit_tests/gtests/test_generic_op.cpp
@@ -1308,6 +1308,10 @@ TEST_F(MeshDevice1x4FabricFixture, TestGenericOpAllGather) {
 
         // Writer CT args
         std::vector<uint32_t> writer_ct_args = common_ct_args;
+        // Keep this manual descriptor in sync with minimal_default_writer.cpp. The writer
+        // consumes this argument before the optional worker-mux configuration; the reader
+        // does not consume it, so it must not be part of common_ct_args.
+        writer_ct_args.push_back(ring_size - 1);  // barrier_target_count
         // fabric_mux_connection_ct_args
         writer_ct_args.push_back(mux_config.get_num_buffers(tt::tt_fabric::FabricMuxChannelType::FULL_SIZE_CHANNEL));
         writer_ct_args.push_back(

